packages/toolkit/src/view/knowledge/components/FileDetailsOverlay.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { Dialog, Icons, ScrollArea, Skeleton } from '@instill-ai/design-system';
+import { useGetFileContent, useListChunks } from '../../../lib/react-query-service/knowledge';
+
+type FileDetailsOverlayProps = {
+    fileUid: string;
+    accessToken: string | null;
+    onClose: () => void;
+    kbId: string;
+    showFullFile: boolean;
+    selectedChunkUid?: string;
+    ownerId: string;
+    isOpen: boolean;
+    setIsOpen: (open: boolean) => void;
+    fileName: string;
+    highlightChunk?: boolean;
+    fileType: string;
+};
+
+const FileDetailsOverlay = ({
+    fileUid,
+    accessToken,
+    onClose,
+    kbId,
+    showFullFile,
+    selectedChunkUid,
+    ownerId,
+    isOpen,
+    setIsOpen,
+    fileName,
+    highlightChunk = false,
+    fileType
+}: FileDetailsOverlayProps) => {
+    const { data: fileContent, isLoading: isLoadingContent } = useGetFileContent({
+        fileUid,
+        kbId,
+        accessToken,
+        enabled: isOpen,
+        ownerId,
+    });
+
+    const { data: chunks, isLoading: isLoadingChunks } = useListChunks({
+        kbId,
+        accessToken,
+        enabled: isOpen && highlightChunk,
+        ownerId,
+        fileUid,
+    });
```

Comment:
@thewbuk Please don't destruct the react-query return, or you need to keep casting new name for each state